docs/src/modules/spring/pages/using-acls.adoc

```diff
@@ -0,0 +1,192 @@
+
+Here is an example ACL on a method:
+
+```java
+    @PostMapping
+    @Acl(
+       allow = @Acl.Matcher(service = "*"), 
+       deny =  @Acl.Matcher(service = "my-service"))
+    public Effect<String> createUser(@RequestBody CreateEmployee create) {
+     ...
+    }
+```
+
+The above ACL allows traffic to all services except the service called `my-service`. 
+
+To allow all traffic:
+
+```java
+@Acl(allow = @Acl.Matcher(principal = Acl.Principal.ALL) 
+```
+
+To allow only traffic from the internet:
+
+```java
+@Acl(allow = @Acl.Matcher(principal = Acl.Principal.INTERNET) 
+```
+
+To allow traffic from `service-a` and `service-b`:
+
+```java
+ @Acl(allow = {
+        @Acl.Matcher(service = "service-a"), 
+        @Acl.Matcher(service = "my-service")}) 
+```
+
+To block all traffic, an ACL with no allows can be configured:
+
+```java
+ @Acl(allow = {})
+```
+
+=== Sharing ACLs between methods
+
+The above examples show how to configure an ACL on a method. ACLs can also be shared between all methods on a service by specifying them on the class, at type level:
+
+```java
+@Acl(allow = @Acl.Matcher(principal = Acl.Principal.INTERNET))
+public class MyAction extends Action { ... } 
+```
+
+The service's ACL can be overridden by individual methods, by specifying the ACL on the method. Note that an ACL defined on a method completely overrides an ACL defined on a service. It does not add to it. So for example, in the following service:
+
+```java
+@Acl(allow = @Acl.Matcher(service = "service-a"))
+ public class MyAction extends Action {
+    @PostMapping
+    @Acl(allow = @Acl.Matcher(service = "service-b"))
+    public Effect<String> createUser(@RequestBody CreateEmployee create) {
+      return ...;
+    }
+}
+```
+
+The `MyMethod` rpc method will allow calls by `service-b`, but not by `service-a`.
+
+=== Configuring the default policy
+
+The default policy can be configured by specifying a project level annotation in the `Main`, for example, to set a default policy of allowing all local services:
+
+```java
+@Acl(allow = @Acl.Matcher(service = "*"))
+public class Main { ... } 
+```
+
+An ACL declared at the project level is used as the default for all services that don't declare their own explicit ACL.
+
+=== Customizing the deny code
+
+When a request is denied, by default, a gRPC error code of 7, `PERMISSION_DENIED`, is sent. This gets transcoded to an HTTP status code of 403, forbidden. The code that is returned when a request is denied can be customised using the `deny_code` property. The deny code must be a valid gRPC error code, which is any integer from 1 to 16, described https://grpc.github.io/grpc/core/md_doc_statuscodes.html[here]. Each one maps as follows to an https://doc.akka.io/japi/akka-http/current/akka/http/scaladsl/model/StatusCodes.html[Akka HTTP status code].
```

Comment:
The user will be better off not knowing about gRPC codes. I put some thoughts here https://github.com/lightbend/kalix/issues/7901

Reply:
We should go with that for now. There is already a tickets for that:
https://github.com/lightbend/kalix/issues/7743

Reply:
no need :)